### PR TITLE
*: various windows fixes

### PR DESCRIPTION
**Pull request by tamird**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/14545)

### CODE REVIEW

File: pkg/util/caller/resolver.go [hunk 1]
```diff
@@ -120,7 +119,7 @@ func (cr *CallResolver) Lookup(depth int) (file string, line int, fun string) {
 		if len(matches) == 1 {
 			file = matches[0]
 		} else {
-			file = filepath.Join(matches[1:]...)
+			file = path.Join(matches[1:]...)
```
Comment (RaduBerinde) [inline pkg/util/caller/resolver.go]: The `path` vs `filepath` distinction can be a bit sublte. Can you add a comment explaining that `path.Join` always uses slash separators (as opposed to `filepath.Join`)?

Comment (tamird) [inline pkg/util/caller/resolver.go]: I'm not sure it's appropriate to add that comment; this is patently the difference between the two packages.

If I added this comment, where exactly would you like to see it?

Comment (RaduBerinde) [inline pkg/util/caller/resolver.go]: Right above this call.

**Comment by RaduBerinde** [comment]
LGTM

File: pkg/util/caller/resolver.go (hunk 1 above)
Comment (RaduBerinde) [inline pkg/util/caller/resolver.go]: Can just be a simple `// Join the path using unix separators.` or whatever

Comment (tamird) [inline pkg/util/caller/resolver.go]: Done.

**Comment by RaduBerinde** [comment]
Thanks, LG!

**Review by bdarnell — APPROVED** [review]

**Pull request merged**